src/SonarScanner.MSBuild.Shim/SonarScanner.Wrapper.cs

```diff
@@ -89,28 +90,52 @@ private static bool InternalExecute(AnalysisConfig config, IEnumerable<string> u
                 return false;
             }
 
-            var exeFileName = FindScannerExe();
+            var exeFileName = FindScannerExe(logger, Path.GetDirectoryName(typeof(SonarScannerWrapper).Assembly.Location));
             return ExecuteJavaRunner(config, userCmdLineArguments, logger, exeFileName, fullPropertiesFilePath, new ProcessRunner(logger));
         }
 
-        private static string FindScannerExe()
+        internal /* for testing */ static string FindScannerExe(ILogger logger, string scannerCliDirectoryLocation, string scannerVersion = SonarScannerVersion)
         {
-            var binFolder = Path.GetDirectoryName(typeof(SonarScannerWrapper).Assembly.Location);
-            var fileExtension = PlatformHelper.IsWindows() ? ".bat" : "";
-            return Path.Combine(binFolder, $"sonar-scanner-{SonarScannerVersion}", "bin", $"sonar-scanner{fileExtension}");
+            var scannerCliFolder = Path.Combine(scannerCliDirectoryLocation, $"sonar-scanner-{scannerVersion}");
+
+            if (!Directory.Exists(scannerCliFolder))
+            {
+                // We unzip the scanner-cli-{version}.zip while in the user's machine so that the Unix file permissions are not lost.
+                // The unzipping happens only once, during the first scanner usage.
+                var zipPath = Path.Combine(scannerCliDirectoryLocation, $"sonar-scanner-cli-{scannerVersion}.zip");
+                logger.LogInfo($"Unzipping sonar-scanner-cli-{scannerVersion}.zip");
+                if (!File.Exists(zipPath))
+                {
+                    logger.LogError($"Could not find {zipPath}");
```

Comment:
I agree with @csaba-sagi-sonarsource 's point here https://github.com/SonarSource/sonar-scanner-msbuild/pull/1250/files#r895532981

It doesn't make sense to call `ExtractToDirectory` if you know it will fail (it will be confusing to read)

so here it should either
- be a thrown exception from the program with the message "Could not find zipPath - the scanner-cli is necessary for the END step" - like this the error will be clear
- or add an `if/else` and rely on the invoking commandline to fail given it's a logged error - however intuitively it might be better to throw an exception